src/base/os.lua

```diff
@@ -80,6 +80,9 @@
 			if os.is("macosx") then
 				formats = { "lib%s.dylib", "%s.dylib" }
 				path = os.getenv("DYLD_LIBRARY_PATH") or ""
+			elseif os.is("cygwin") then
+				formats = { "lib%s.a", "%s.a", "lib%s.dll.a", "%s.dll.a" }
+				path = os.getenv("LD_LIBRARY_PATH") or ""

```

Comment:
I would much rather introduce a os.is("posix") that way it would cover any posix compliant OS... cygwin, mingw, are not the only "unix" based OS's out there... the "os.is" function could easily have this list internally, and return true on "posix" instead..